src/Arena.js

```diff
@@ -1,554 +1,555 @@
+  /*---------------------PASSIVES, BUFFS AND BUFF DURATIONS---------------------------*/
 
-  //Riposte function will activate during opponent turn (on successful block)
-  async function riposteAttack() {
-    await sleep(200)
-    let anim = document.createElement("img")
-    myHit = diceRoll(myGladiator.maxDmg) + dmgBonus + toHitBonus;
-    dispatch(riposte(myHit))
-    animatePlayerAttack(myGladiator.name)
-    displayDmgNumber(myHit, "skill-number")
-    anim.className = 'skill-animation'
-    anim.src= require(`./assets/images/animations/riposte.gif`)
-    document.getElementsByClassName('center-screen')[0].appendChild(anim)
-    setTimeout(() => {  anim.parentNode.removeChild(anim); }, 500)
+  function animateAttack(dualWield){
+    setAttackAnimation({enabled: true, dualWield});
+    setTimeout(() => {
+      setAttackAnimation({enabled: false})
+    }, 300)
   }
 
-  if ('rage' in activeBuffs) {
-    if (stack > 5) {
-      stack = 5
-    }
-    dmgBonus += stack
-  }
+  //Riposte function will activate during opponent turn (on successful block)
 
-  if ('champion' in activeBuffs) {
-    dmgBonus += 3
-    myAC += 3
-    critChance += 20
-    toHitBonus += 3
-    blockValue += 3
+  async function riposteAttack() {
+    await sleep(200);
+    setRiposteAnimation(true);
+    const myHit = diceRoll(gladiator.maxDmg) + gladiator.dmgBonus + gladiator.toHitBonus;
+    dispatch(riposte(myHit));
+    animateAttack();
+    displayDmgNumber(myHit, "skill-number");
+    setTimeout(() => {
+      setRiposteAnimation(false);
+    }, 500);
   }
 
   // Decrements buff durations every time player turn is activated
-  function decrementBuffDuration() {
-    if ('rage' in activeBuffs) {
-      activeBuffs.rage.duration -= 1
-      if (activeBuffs.rage.duration === 0){
-        document.getElementById("weaponBuff").classList.remove("rage-active")
-        delete activeBuffs.rage
-      }
-    }
-    if ('riposte' in activeBuffs){  
-      activeBuffs.riposte.duration -=1
-      if (activeBuffs.riposte.duration === 0){
-        document.getElementById("weaponBuff").classList.remove("riposte-active")
-        delete activeBuffs.riposte
-      }
-    }
-    if ('poisonTrident' in activeBuffs){
-      activeBuffs.poisonTrident.duration -=1
-      if (activeBuffs.poisonTrident.duration === 0){
-        document.getElementById("weaponBuff").classList.remove("poisonTrident-active")
-        delete activeBuffs.poisonTrident
-      }
-    }
-    if ('champion' in activeBuffs){
-      activeBuffs.champion.duration -=1
-      if (activeBuffs.champion.duration === 0){
-        document.getElementById("weaponBuff").classList.remove("champion-active")
-        delete activeBuffs.champion
-      }
-    }
-  }
 
-/*---------------------COMBAT---------------------------*/
-  
+  /*---------------------COMBAT---------------------------*/
+
+  useEffect(() => {
+    gladiator.updateGladiator(myGladiator);
```

Comment:
Any time the redux value of `myGladiator` changes it will update stuff here - this is probably all broken due to how gladiator state could/should be separated from "baseline" and "arena instance"